### ISSUE TITLE
Ubuntu 22.04: Option "click-action: minimize-or-cycle" needed!

### ISSUE THREAD

**Issue by meddev-code** [micheleg/dash-to-dock#1749]
If there is a single window -> minimize
else -> cycle

self explanatory
thank You!

-----

>             case clickAction.MINIMIZE_OR_CYCLE:
>                 //MINIMIZE OR CYCLE  modification by s1w_
>                 if (!Main.overview.visible) {
>                     if (this.focused && !hasUrgentWindows && windows.length > 1)
>                        //(windows.length > 1 || modifiers || button != 1))
>                         this._cycleThroughWindows();
>                     else {
>                         let w = windows[0];
>                         if (this.focused) {
>                             // Window is raised, minimize it
>                             this._minimizeWindow(w);
>                         } else {
>                             // Window is minimized, raise it
>                             Main.activateWindow(w);
>                         }
>                     }
>                 }
>                 else
>                     this.app.activate();
>                 break;

**Comment by fabd** [comment]
Another vote for "Cycle windows + minimize" as available in Dash To Panel.

This one feels the most convenient for me, I often have two terminals side by side, and my main app like browser or VSCode in the center. If I switch to terminal (cmd + 3) then in Dash 2 Panel it would just focus the last active Terminal window, and if I press Cmd +3 again, it would simply focus the next terminal window. There is never a need to have the previews dialog or the fullscreen switch to overview.

**Comment by fabd** [comment]
Thanks @s1ww it works! Had to modify the existing `case clickAction.FOCUS_MINIMIZE_OR_PREVIEWS:` as I'm not sure where to add the new option.

With this change it appears to behave exactly like in Dash To Panel, perfect. I'm able to quickly switch between eg. my two terminals and chrome :+1: 

To be fair I can see how it may not be intuitive for other users - if you have like 3 terminals, then when you switch back from chrome to termianl, and say you want to cycle to the next terminal, then the order in which it cycles is not fixed. As you focus back from another app, the order changes.

In this case (3+ windows for an app are open) I find that even using "Show next application window" is annoying too.

But for me , as I always use only two terminal windows, the "cycle windows + minimize" works great, because the "cycle" part essentialy is just a toggle between two windows of the same app, therefore it's really obvious where the focus is going.

This makes me wonder exactly how Windows 10 handled this. 

If the "cycling through windows" always followed the same order, perhaps the order in which the windows were created, then after I switch back from eg. browser, to my terminals, then it would also be obvious where the focus goes since I could just open my terminals in visual order from left to right, and cycling windows would always go in the same direction.. but for whatever reason it doesn't.

**Comment by fabd** [comment]
For science I checked again in Dash To Panel,

to be exact, this code above corresponds to `Cycle through windows`, it doesn't minimize an app with 2+ windows.

What the issue title refers to would be another option in Dash To Panel `Cycle + minimize`

The second once, does minimize the app, after you cycled through the windows - which is handy.

The first one, and the code above, indefinitely cycles through the windows, once focused - and then you have to use global shortcut like Cmd H if you want to minimize but then it can be annoying as it only minimizes one window and not the whole app's windows.

So long story short if only one is implemented I think "Cycle + minimize" is the most flexible - since if you want to cycle indefinitely while in the same app, you can use the "Show next application window" shortcut. The "Cycle + minimize'" usefulness is more like, you switch back from app A to app B, but you want the _other_ window of app B. So you just tap once more the shortcut.

**Comment by vanvugt** [comment]
Duplicate of #365
